### PR TITLE
Integration tests for plotter service

### PR DESCRIPTION
**Pull request by luisferreira32**
It is important to ensure the expected functioning of the plotter service
when developing. This integration tests is a first step in setting up a CI
that will automate this tests for us.

### CODE REVIEW

File: services/plotter/plotter.py [hunk 1]
```diff
@@ -6,9 +6,6 @@
 
 
 def plotter(data, configs):
-    # logger.debug("Building plot for given data.")
-    fig = plt.figure()
```
Comment (luisferreira32) [inline services/plotter/plotter.py:10]: flake8 was complaining here, since `fig` was unused...?

File: services/plotter/server.py [hunk 2]
```diff
@@ -18,8 +18,8 @@ def do_POST(self):
         except (InvalidRequestError, ValidationError, ValueError) as e:
             response = "Bad Request: " + str(e)
             self._return_400(response)
-        except Exception:
-            response = "Oops: something went wrong"
+        except Exception as e:
+            response = "Oops: something went wrong...\n" + str(e)
```
Comment (luisferreira32) [inline services/plotter/server.py:22]: Felt like we can add a bit more info for now to the caller response on 500

File: services/utils/server_handler.py [hunk 3]
```diff
@@ -19,13 +19,13 @@ def _return_405(self, message: str = "Method Not Allowed"):
         """Sends a response with status code 405 - Method Not Allowed"""
         self.send_response(405)
         self._set_headers(len(message))
-        self.wfile.write(bytes(message))
+        self.wfile.write(message.encode())
```
Comment (luisferreira32) [inline services/utils/server_handler.py:22]: This fix had been missing for a while!

File: tests/integration/plotter_test.py [hunk 4]
```diff
@@ -0,0 +1,37 @@
...
+            files={
+                'file': ('test.csv', data),
+                'options': '{"title": "abc"}'
+            }
+        )
+
+        # expect a 200 and a defined plot
+        self.assertEqual(response.status_code, 200)
+
+        # TODO compare with reference image (but not by pixel)
```
Comment (luisferreira32) [inline tests/integration/plotter_test.py:25]: Note that matplotlib is actually pretty flaky on giving you a consistent response, just in terms of bytes I manage to get variations of about 200. So I decided to leave this as a TODO - some image comparison that is flexible enough. Maybe you can propose something in a follow-up PR?

**Comment by luisferreira32** [comment]
Another note is that the `flake8` is still not passing since we have a lot of commented code. I'll open a non-feature PR later to just clean up linting and ensure we got some bugs fixed.

File: services/plotter/plotter.py (hunk 1 above)
Comment (MrToino) [reply services/plotter/plotter.py:10]: Yes, it is not yet used, cuz later on I'll be implementing a different way to get the Matplotlib figure. It is a bit weird how it works, but you'll get it I hope.
But for now, you can just take out the `fig`.

File: services/plotter/server.py (hunk 2 above)
Comment (MrToino) [reply services/plotter/server.py:22]: Yes agree

**Review by MrToino — CHANGES_REQUESTED** [review]
I believe these test .png files should stay stored locally. Please make sure to add that to the .gitignore file.

File: services/utils/server_handler.py (hunk 3 above)
Comment (MrToino) [reply services/utils/server_handler.py:22]: That's good you noticed it!

**Comment by luisferreira32** [comment]
Let's merge the #33 first and adapt this one after that. On another note, the png file belongs here. It's a reference file for the automated tests that will be running on the future git workflows CI. (we're just missing the implementation to use it)

**Review by MrToino — APPROVED** [review]
That's good to go.

**Pull request merged**